### PR TITLE
fix: swagger details for apis

### PR DESCRIPTION
**Pull request by jainpawan21**

### CODE REVIEW

File: apps/api/src/app/shared/framework/paginated-ok-response.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { ApiExtraModels, ApiOkResponse, getSchemaPath } from '@nestjs/swagger';
```
Comment (jainpawan21) [inline apps/api/src/app/shared/framework/paginated-ok-response.decorator.ts:1]: taken from this blog
https://aalonso.dev/blog/how-to-generate-generics-dtos-with-nestjsswagger-422g

File: apps/api/src/app/shared/dtos/pagination-request.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    type: Number,
+    required: false,
+  })
+  page?: number = 0;
+
+  @ApiPropertyOptional({
+    type: Number,
+    required: false,
+    default: 10,
+    maximum: 100,
```
Comment (p-fernandez) [inline apps/api/src/app/shared/dtos/pagination-request.ts:14]: One problem with this approach is that different pagination endpoints might have different default and maximum values. We must think in a way to provide those values dynamically.

**Review by p-fernandez — APPROVED** [review]
This will be a great improvements for our API docs!
🌟

File: apps/api/src/app/shared/framework/paginated-ok-response.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { PaginatedResponseDto } from '../dtos/pagination-response';
+import { Type, applyDecorators } from '@nestjs/common';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const ApiOkResponsePaginated = <DataDto extends Type<unknown>>(dataDto: DataDto) =>
```
[the remark this answers is not in the document]
Comment (jainpawan21) [reply apps/api/src/app/shared/framework/paginated-ok-response.decorator.ts]: Done

File: apps/api/src/app/shared/dtos/pagination-request.ts (hunk 2 above)
Comment (jainpawan21) [reply apps/api/src/app/shared/dtos/pagination-request.ts]: Done

**Pull request merged**